src/NuGet.Core/NuGet.ProjectModel/Utf8JsonStreamPackageSpecReader.cs

```diff
@@ -0,0 +1,1873 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Text.Json;
+using NuGet.Common;
+using NuGet.Configuration;
+using NuGet.Frameworks;
+using NuGet.LibraryModel;
+using NuGet.Packaging.Core;
+using NuGet.RuntimeModel;
+using NuGet.Versioning;
+
+namespace NuGet.ProjectModel
+{
+    internal static class JsonStreamPackageSpecReaderPropertyName
+    {
+        private static readonly char[] VersionSeparators = new[] { ';' };
```

Comment:
> I think using PropertyName suffix, rather than Utf8 prefix, will make this easier to understand. Down in the Read method, it will be if (reader.ValueTextEquals(LevelPropertyName)) rather than if (reader.ValueTextEquals(Utf8Level))
> 
> I can't remember where I first saw this pattern, but another option is to make a private static class called PropertyNames. For example:
> 
> internal class UtfJsonStreamAssetsLogMessageConverter : ...
> {
>   public override AssetsLogMessage Read(ref Utf8JsonStreamReader reader)
>   { 
>     // stuff
>     if (reader.ValueTextEquals(PropertyName.Level)
>    // other stuff
>   }
> 
>   private static class PropertyName
>   {
>     public static readonly byte[] Level = Encoding.UTF8.GetBytes(LogMessageProperties.LEVEL);
>     // other properties
>   }
> }
- @zivkan https://github.com/NuGet/NuGet.Client/pull/5529#discussion_r1415105618